code/modules/hydroponics/plant_genes.dm

```diff
@@ -0,0 +1,198 @@
+/datum/plant_gene/trait
+	var/rate = 0.05
+	var/examine_line = ""
+
+/datum/plant_gene/trait/Copy()
+	var/datum/plant_gene/trait/G = ..()
+	G.rate = rate
+	return G
+
+/datum/plant_gene/trait/proc/on_consume(obj/item/weapon/reagent_containers/food/snacks/grown/G, mob/living/carbon/target)
+	return
+
+/datum/plant_gene/trait/proc/on_slip(obj/item/weapon/reagent_containers/food/snacks/grown/G, mob/living/carbon/target)
+	return
+
+/datum/plant_gene/trait/proc/on_squash(obj/item/weapon/reagent_containers/food/snacks/grown/G, atom/target)
+	return
+
+
+/datum/plant_gene/trait/squash
+	// Allows the plant to be squashed when thrown or slipped on, leaving a colored mess and trash type item behind.
+	// Also splashes everything in target turf with reagents and applies other trait effects (teleporting, etc) to the target by on_squash.
+	// For code, see grown.dm
+	name = "Liquid Contents"
+	examine_line = "<span class='info'>It has a lot of liquid contents inside.</span>"
+
+/*/datum/plant_gene/trait/squash/on_slip(obj/item/weapon/reagent_containers/food/snacks/grown/G, mob/living/carbon/target)
+	G.squash(target)*/
+
+
+/datum/plant_gene/trait/slip
+	// Makes plant slippery, unless it has a grown-type trash. Then the trash gets slippery.
+	// Applies other trait effects (teleporting, etc) to the target by on_slip.
+	// For code, see grown.dm
+	name = "Slippery Skin"
+	rate = 0.1
+	examine_line = "<span class='info'>It has a very slippery skin.</span>"
+
+
+/datum/plant_gene/trait/cell_charge
+	// Cell recharging trait. Charges all mob's power cells to (potency*rate)% mark when eaten.
+	// Generates sparks on squash.
+	// Small (potency*rate*5) chance to shock squish or slip target for (potency*rate*5) damage.
+	// Multiplies max charge by (rate*1000) when used in potato power cells.
+	name = "Electrical Activity"
+	rate = 0.01
+
+/datum/plant_gene/trait/cell_charge/on_slip(obj/item/weapon/reagent_containers/food/snacks/grown/G, mob/living/carbon/C)
+	var/power = G.seed.potency*rate*5
+	if(prob(power))
+		C.electrocute_act(round(power), G, 1, 1)
+
+/datum/plant_gene/trait/cell_charge/on_squash(obj/item/weapon/reagent_containers/food/snacks/grown/G, atom/target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		var/power = G.seed.potency*rate*5
+		if(prob(power))
+			C.electrocute_act(round(power), G, 1, 1)
+
+/datum/plant_gene/trait/cell_charge/on_consume(obj/item/weapon/reagent_containers/food/snacks/grown/G, mob/living/carbon/target)
+	if(!G.reagents.total_volume)
+		var/batteries_recharged = 0
+		for(var/obj/item/weapon/stock_parts/cell/C in target.GetAllContents())
+			var/newcharge = min(G.seed.potency*0.01*C.maxcharge, C.maxcharge)
+			if(C.charge < newcharge)
+				C.charge = newcharge
+				if(isobj(C.loc))
+					var/obj/O = C.loc
+					O.update_icon() //update power meters and such
+				C.update_icon()
+				batteries_recharged = 1
+		if(batteries_recharged)
+			target << "<span class='notice'>Your batteries are recharged!</span>"
```

Comment:
to be fair, it's not the recharging that's OP here - it's the fact that you can fit 7 guns in your backpack in the first place.